posts/2022-09-01-22.0.0.10-beta.adoc

```diff
@@ -0,0 +1,581 @@
+
+The REST handler framework references the BELL service to serve the `/bellEP` endpoint. During the first reference the server creates an instance of class BellEndpoint and injects an unmodifiable map with key/value pair `"hello"/"WORLD"`. The server injects BELL properties at service creation and after BELL configuration updates.
+
+You can find out more about this specific update from the following links:
+
+* link:https://openliberty.io/docs/latest/reference/feature/bells-1.0.html[Open Liberty BELL 1.0 Feature Doc]
+* link:https://docs.oracle.com/javase/9/docs/api/java/util/ServiceLoader.html[Java ServiceLoader]
+* link:https://openliberty.io/docs/latest/reference/config/bell.html[Open Liberty BELL Server Configuration Doc]
+
+// https://github.com/OpenLiberty/open-liberty/issues/21847
+[#microprofile]
+== MicroProfile Telemetry 1.0
+
+link:https://opentelemetry.io/[[OpenTelemetry] is a set of APIs, SDKs, tooling and integrations that are designed for the creation and management of telemetry data such as traces, metrics, and logs.
+
+This feature adopts OpenTelemetry from CNCF to allow MicroProfile Applications to benefit from OpenTelemetry and enable automatic tracing of Jakarta RESTful Web Services if configured.
+
+[source, xml]
+----
+<featureManager>
+  <feature>mpTelemetry-1.0</feature>
+</featureManager>
+----
+
+You can view the telemetry data from your application in link:https://www.jaegertracing.io/docs/1.37/monitoring/#traces[Jaeger] and  link:https://zipkin.io/pages/quickstart.htmlp[Zipkin]. To export your traces to one of these exporters, configure the exporter with system properties, e.g. `-Dotel.traces.exporter=jaeger` or environment variables, e.g. `OTEL_TRACES_EXPORTER=jaeger`.  You can also configure the exporter endpoints using `OTEL_EXPORTER_JAEGER_ENDPOINT` and `OTEL_EXPORTER_ZIPKIN_ENDPOINT`.  To specify the name of the service, use `OTEL_SERVICE_NAME`.
+   
+For more information about MicroProfile Telemetry, see the following links:
+
+* link:https://github.com/eclipse/microprofile-telemetry[microprofile-telemetry]
+* link:https://github.com/open-telemetry/opentelemetry-specification/blob/v1.11.0/specification/trace/api.md[opentelemetry-specification]
+* link:https://opentelemetry.io[opentelemetry.io]
+
+// https://github.com/OpenLiberty/open-liberty/issues/20209
+[#security]
+== Distributed Security Caches
+
+   
+   - A sentence or two that introduces the update to someone new to the general technology/concept.
+   
+ > Distributed security cache support has been introduced so that multiple Liberty servers can share caches via a JCache provider.
+
+   - What was the problem before and how does your update make their life better? (Why should they care?)
+
+> Prior to this feature, the authentication (subject) and logged out cookie caches were restricted to be local and in-memory. Multiple servers were unable to benefit from their peers' caches and each server started with a cold cache.

```

Comment:
Can we give a more concrete example of why servers starting with a cold cache is a problem that needed solving? From this I don't get a sense of why I should care about distributed security caches. Yet, I know they're useful because we had this nice blog post in the past about persistence using JCache: https://openliberty.io/blog/2018/03/22/distributed-in-memory-session-caching.html Having a concrete example like a shopping trolley makes it clearer what the benefit is. How would starting with a cold cache be a problem in that example (if relevant; if not, is there a better example for this situation?)?

Reply:
Removed